src/Extensions/Parser.hs

```diff
@@ -0,0 +1,105 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Parser for Haskell Modules to get all Haskell Language Extensions used.
+-}
+
+module Extensions.Parser
+       ( parseFilePath
+       , parseSourceFile
+
+         -- * Helper types
+       , ExtensionsParseResult
+       ) where
+
+import Data.ByteString (ByteString)
+import GHC.LanguageExtensions.Type (Extension (..))
+import Text.Parsec (ParseError, alphaNum, between, char, eof, many, many1, manyTill, oneOf, parse,
+                    sepBy1, try, unexpected, (<|>))
+import Text.Parsec.ByteString (Parser)
+import Text.Parsec.Char (anyChar, endOfLine, letter, space, spaces, string)
+import Text.Read (readMaybe)
+
+import qualified Data.ByteString as BS
+
+
+-- | Type alias for the resulting type of module parsing.
+type ExtensionsParseResult = Either ParseError [Extension]
+
+{- | By the given file path, reads the file and returns parsed list of
+'Extension's, if parsing succeeds.
+-}
+parseFilePath :: FilePath -> IO (Either ParseError [Extension])
+parseFilePath file = parseSourceFile <$> BS.readFile file
+
+{- | By the given file source content, returns parsed list of
+'Extension's, if parsing succeeds.
+-}
+parseSourceFile :: ByteString -> Either ParseError [Extension]
+parseSourceFile = parse extensionsP "SourceName"
+
+{- | The main parser of 'Extension's.
+
+It parses language pragmas or comments until end of file or the first line with
+the function/import/module name.
+-}
+extensionsP :: Parser [Extension]
+extensionsP = concat <$> manyTill
```

Comment:
I thought to do that when we will merge extensions from a module and .cabal file. :thinking:

Reply:
@vrom911 Great idea :+1: I also haven't done in my PR. I agree with you, that instead of doing it 3 times (parsing, cabal, merging), we can do it only once in merging.